ccmlib/scylla_node.py

```diff
@@ -1333,18 +1333,19 @@ def rollback(self, upgrade_to_version):
     def watch_rest_for_alive(self, nodes, timeout=120):
         """
         Use the REST API to wait until this node detects that the nodes listed
-        in "nodes" become fully operational and knows of its tokens.
+        in "nodes" become fully operational as normal token owners.
         This is similar to watch_log_for_alive but uses ScyllaDB's REST API
         instead of the log file and waits for the node to be really useable,
         not just "UP" (see issue #461)
         """
         logging.getLogger('urllib3.connectionpool').disabled = True
         try:
-            tofind = nodes if isinstance(nodes, list) else [nodes]
-            tofind = set([node.address() for node in tofind])
+            nodes_tofind = nodes if isinstance(nodes, list) else [nodes]
+            tofind = set([node.address() for node in nodes_tofind])
+            tofind_host_id_map = dict([(node.address(), node.node_hostid if hasattr(self, 'node_hostid') else None) for node in nodes_tofind])
```

Comment:
I think we can retrieve the host_id via the api if it's yet available (the ccmlib node.py method uses nodetool that is it ready yet since scylla-jmx was not started yet)